config.go

```diff
@@ -571,3 +588,24 @@ func netName(chainParams *netparams.Params) string {
 	}
 	return chainParams.Name
 }
+
+// retrieveRootPath drops all extra characters that are not part of the root path.
+// i.e. with input http://www.mydomain.com/xxxxx, http://www.mydomain.com should
+// be returned.
+func retrieveRootPath(path string) (string, error) {
+	r, err := url.Parse(path)
+	if err != nil {
+		return "", fmt.Errorf("invalid '%s' url used. error: %v", path, err)
+	}
+
+	// If the url scheme was not found, a regex expression can be used to
+	// eliminate the unwanted part.
+	if r.Scheme == "" {
+		exp := regexp.MustCompile(`([\/?]\S*)`)
```

Comment:
Generally we should make tests for our regexps since they are so full of gotchas.  This is ok for here though.